README.md

```diff
@@ -1,14 +1,77 @@
 # webpack-partial
 
-Intelligently merge webpack configuration files.
+Intelligently compose [webpack] configuration files.
 
 ![build status](http://img.shields.io/travis/webpack-config/webpack-partial/master.svg?style=flat)
 ![coverage](http://img.shields.io/coveralls/webpack-config/webpack-partial/master.svg?style=flat)
 ![license](http://img.shields.io/npm/l/webpack-partial.svg?style=flat)
 ![version](http://img.shields.io/npm/v/webpack-partial.svg?style=flat)
 ![downloads](http://img.shields.io/npm/dm/webpack-partial.svg?style=flat)
 
-Take a multiple webpack configurations (or functions generating webpack configurations) and merge them. Usually used in the manner of `partial(fullConfig, partialConfig)` to return a new complete configuration.
+
+## Usage
+
+```sh
+npm install --save webpack-partial
+```
+
+Making [webpack] configurations composable is tricky. So we provide several helpers to ease the composition process. Each helper is of the form `(arg1, arg2, ..., webpackConfig)`. All functions are curried and the last argument is always a webpack configuration object allowing you to chain helpers together easily with `compose`.
+
+Simple single-plugin example:
+
+```javascript
+import plugin from 'webpack-partial/plugin';
+import StatsWebpackPlugin from 'stats-webpack-plugin';
+
+const config = plugin(new StatsWebpackPlugin(), {/* webpack config here */});
+```
+
+Now with multiple plugins:
+
+```javascript
+import plugin from 'webpack-partial/plugin';
+import StatsWebpackPlugin from 'stats-webpack-plugin';
+import ExtractTextWebpackPlugin from 'extract-text-webpack-plugin';
+
+const config = plugin(
```

Comment:
The composition of config below is SOOOOOO NICE and is the whole reason I use it. If I were you I'd be tempted not to even show this so you can support composition as the one way of doing things. This may work but composition is 👌 ✨ .